--- src/RedisBloom/src/rebloom.c    2025-07-26 20:24:02.274409291 -0700
+++ rebloom.c    2025-07-27 00:21:37.953346863 -0700
@@ -372,7 +372,7 @@
             if (++cur_pos == argc) {
                 return RedisModule_WrongArity(ctx);
             }
-            if (tolower(*(argstr + 1)) == 'r') { // error rate
+            if (arglen >= 2 && tolower(*(argstr + 1)) == 'r') { // error rate
                 if (RedisModule_StringToDouble(argv[cur_pos++], &options.error_rate) !=
                         REDISMODULE_OK ||
                     !isConfigValid(options.error_rate, rm_config.bf_error_rate)) {
@@ -403,7 +403,7 @@
             break;
 
         case 'n':
-            if (tolower(*(argstr + 2)) == 'c') {
+            if (arglen >= 3 && tolower(*(argstr + 2)) == 'c') {
                 options.autocreate = 0;
             } else {
                 options.nonScaling = BLOOM_OPT_NO_SCALING;
